ethclient/ethclient.go

```diff
@@ -76,6 +76,12 @@ type rpcBlock struct {
 	UncleHashes  []common.Hash    `json:"uncles"`
 }
 
+func (ec *Client) BlockNumber(ctx context.Context) (big.Int, error) {
```

Comment:
It seems the go-ethereum team are [against adding this](https://github.com/ethereum/go-ethereum/pull/15907), should we take a different approach as suggested?

Reply:
Hmm, annoying. Thanks for catching this.

Reply:
Are we going with a different approach?